plugin/core/sessions.py

```diff
@@ -257,6 +257,56 @@ def get_initialize_params(variables: Dict[str, str], workspace_folders: List[Wor
             },
             "codeLens": {
                 "dynamicRegistration": True
+            },
+            "semanticTokens": {
+                "dynamicRegistration": True,
+                "tokenTypes": [
+                    "namespace",
+                    "type",
+                    "class",
+                    "enum",
+                    "interface",
+                    "struct",
+                    "typeParameter",
+                    "parameter",
+                    "variable",
+                    "property",
+                    "enumMember",
+                    "event",
+                    "function",
+                    "method",
+                    "macro",
+                    "keyword",
+                    "modifier",
+                    "comment",
+                    "string",
+                    "number",
+                    "regexp",
+                    "operator"
+                ],
+                "tokenModifiers": [
+                    "declaration",
+                    "definition",
+                    "readonly",
+                    "static",
+                    "deprecated",
+                    "abstract",
+                    "async",
+                    "modification",
+                    "documentation",
+                    "defaultLibrary"
+                ],
+                "formats": [
+                    "relative"
+                ],
+                "requests": {
+                    "range": True,
+                    "full": {
+                        "delta": True
+                    }
+                },
+                "multilineTokenSupport": False,
```

Comment:
Is there a reason why we wouln't support multiline tokens? I think it should already work automatically with add_regions for multiline tokens, no?

Reply:
These are simply copied from what VSCode sends to Pylance. I have no idea what it should be accurately in ST.

Reply:
@jfcherng added this to support his pylance server, clangd worked but adding this.

> Could you explain how it would work? As already pointed out, we probably need to initialize _all_ regions keys for each new view in the desired order, and `View.erase_regions` will do nothing in this regard. Also the order should depend on user settings for document highlights, i.e. if it is set to "fill", then it should be drawn below diagnostics, and otherwise on top of diagnostics. I think the regions for semantic tokens can always be drawn at the bottom (i.e. its region key initialized first).

key used for diagnostic/doc-highlight drawing was a static string. but if you used a new key for every draw call and store that key and used it in clear call, diagnostic/highlighting should draw on top on any 'add_region'.

It works so far I have tested I didn't face any issue drawing diagnostics/highlighting. It didn't worked before at, only drew on 'non add_region' scopes or if server stopped drawing on typo variable name.
<img width="702" alt="diag_highlighting_underline" src="https://user-images.githubusercontent.com/72250074/122397536-af6afb00-cf96-11eb-93f8-72c7631ef18d.png">
<img width="706" alt="diag" src="https://user-images.githubusercontent.com/72250074/122397577-b72a9f80-cf96-11eb-903a-617f479c3c2d.PNG">
<img width="733" alt="doc_highlight_fill" src="https://user-images.githubusercontent.com/72250074/122397598-beea4400-cf96-11eb-906d-ef2aca3429d1.PNG">